src/CHIPBindings.cc

```diff
@@ -1484,7 +1484,7 @@ hipError_t hipEventDestroy(hipEvent_t Event) {
 
   // instead of destroying directly, decrement refc to 1 and  let
   // StaleEventMonitor destroy this event
-  delete Event;
+  Event->decreaseRefCount("hipEventDestroy");
```

Comment:
If we are refcounting user events then these should also be tracked via `track()` so that `hipEventDestroy` is called once these events are no longer needed